### DISCUSSION TITLE
Terrible Search function of Osmand+

### DISCUSSION THREAD

**Discussion by desputinski** [osmandapp/OsmAnd#18210] [General]
Hi, I've been using Osmand+ for about 7 years and in general, I love the app.
BUT the search function is terrible, and much worse than the already bad search function of openstreetmap.org.

For example: It is not possible to find the names of parks in cities. Only street names. And if they are spelled 100% correctly. 
I think both things, not finding misspelled names and not finding names of important places, parks, museums etc. can be called a bug! 

One example: Search for "Richard-Wagner-Hain leipzig". 0 matches! Search for it on Openstreetmap.org: Result: 
https://www.openstreetmap.org/search?query=Richard-Wagner-Hain%20leipzig#map=17/51.33736/12.34824

Best regards
desputin

**Comment by tommycrock** [comment]
Hmm, not all parks
<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/5749884/73ed1e03-03bd-4bb5-8e99-170db0d3ba7a"  width="300" height="auto"/> </td>
  </tr>
</table>
I agree that some ability to cope with misspellings would be good but I don't know how much processing that takes

**Reply by desputinski** [reply]
yes. Another example. The "Stadtpark" ist the largest park in Hamburg/Germany. When I search for "Hamburg Stadtpark" I'm getting all kinds of street names with "Stadtpark" in the name ("Stadtparktreppe", "Am Stadtpark" etc. but not the park itself.

**Reply by tommycrock** [reply]
The search results prioritise roads as in #18125

**Reply by vshcherb** [reply]
Did you try latest search https://osmand.net/map/search/#14/51.3461/12.3395, it has 2 exact results 
<img width="355" alt="Screenshot 2024-11-05 at 10 02 56" src="https://github.com/user-attachments/assets/1c31e160-73bb-48a4-b2c1-7d4b361600a6">. I think the original issue is fixed.

The issue with Hamburg Stadtpark, that it's called "Hamburger Stadtpark". I created an issue https://github.com/osmandapp/OsmAnd/issues/21234

**Comment by desputinski** [comment]
> The search results prioritise roads as in #18125

Hi tommycrock, this is not correct. If I use the "category" seach, Osmand also doesn't find the POI of the park. All roads are listed there too. Sorry to say. The problem remains. It is not the user.

**Reply by tommycrock** [reply]
Well, we have different results. It might be worth deleting the map and redownloading so it reindexes?
I downloaded the Hamburg map and searched for Hamburg Stadtpark, results order depends on where the map is centred but is not dominated by roads:
<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/5749884/31667c17-99e5-47ec-b288-d6a31dc94016"  width="300" height="auto"/> </td>
  </tr>
</table>
Using categories search and selecting the park category I see a list of only parks. If I then start typing stad, I only get two results having centred on the Hamburg area:
<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/5749884/13f979e7-d505-4061-9073-e431f9761ad9"  width="300" height="auto"/> </td>
  </tr>
</table>

**Comment by clamyahoo** [comment]
Hi,
I have a similar problem with supermarkets. I would like to suggest an improvement in the search function: Especially with shops, I would like to be able to just write the name and the town where they are into the search field. For example, there is a REWE supermarket in Vogesenallee 33, 77694 Kehl (it could be anything, Aldi, Edeka, Lidl and other sorts of great supermarkets). In Nominatim I just have to write REWE Kehl and it shows me their three supermarkets in Kehl. In OsmAnd, I can only search by the name or the brand or maybe by the category shop > supermarket. But then it only shows the distances to the next ones or I have to search on the map. This is quite bad. And at the same time, if I look for street, it works, for example if I search "Hauptstrasse Kehl" it shows me the Hauptstraße in Kehl. Would that be possible, e.g. to combine just the name or the brand of an object and the town?
Kind regards

**Reply by vshcherb** [reply]
> (it could be anything, Aldi, Edeka, Lidl and other sorts of great supermarkets)

We just fixed in 4.8-4.9 release as now it displays brand names on top of street names

**Reply by clamyahoo** [reply]
Thank you very much, now, it works!

**Comment by Sonwon1** [comment]
Reverse engineer the search algorithm that Google Maps uses and copy that, it is the best I've seen. Kurviger does a better job too. Just my opinion.

OsmAnd is the best navigation app I have found however some of the tools are a pain to use.

I use to, when plotting a route with OsmAnd use Google Maps to find where and then input the GPS coordinates into OsmAnd because the OsmAnd search was such a pain to use. It was a tedious pain to build a route that way but less so than using OsmAnd search. I have since found a better way that is not a pain.

**Comment by sjvudp** [comment]
I also have an issue when searching for addresses stored in my contacts: Usually I use a line break instead of a comma to format the address, like I would write it on a letter. Unfortunately the line breaks are treated as a space, and therefore the search results are quite bad.

**Reply by clamyahoo** [reply]
Hi,
how do you include addresses in Osmand+? I don't find any option for this and in the settings I can't grant the app permission to access my contact data. Android Version 4.8.6 F-Droid.

**Reply by sjvudp** [reply]
From contacts I long-tap the address, copy it, then start search in OsmAnd and paste the address (I never managed to replace Google maps as default application to display addresses).

**Reply by clamyahoo** [reply]
Thanks a lot, I didn't think so far, but of course, that's it and it works perfectly! If one day you look for ideas to push this further, you can make it possible to display automatically the addresses of one's contacts on the map so that you can for example find more easily the adresses of customers for the navigation if you have to make a tour and choose the best way. But I don't need this function. I've just seen it on another app and thought that this could be useful for this case.

**Comment by clamyahoo** [comment]
Another real improvement would be something that I've found out recently about word separation: There is a school in Offenburg called Hansjakob-Schule. The search function works if I write Hansjakob-Schule or Hansjakob Schule. Then it displays it perfectly and if you're in Offenburg, it will show this as the closest one. 
But if you write Hansjakobschule without Word separation, it won't find it at all. This is quite a problem because at least in german, when people compose words in proper names / titles of Institutionen, museums, companies etc., sometimes, they separate the words, but sometimes not, gramatically, you don't have to. So it could really help if OSMAnd was able to understand it both ways, although I don't know if this is so easy to manage. 
Kind regards,
Clemens

<table style="width:100%">
  <tr>
    <td><img src="https://github.com/user-attachments/assets/3739a929-1075-4b77-a342-2b3c4aa93aaa"  width="300" height="auto"/> </td>
    <td><img src="https://github.com/user-attachments/assets/3ee3b6ce-ff6b-4b71-a6f2-6c17af478d13"  width="300" height="auto"/> </td>
  </tr>
</table>

**Reply by vshcherb** [reply]
Same issue in backend is planned https://github.com/osmandapp/OsmAnd/issues/23709

**Comment by sarahsquirrel** [comment]
Who has ownership of this issue? If no-one has ownership, how can that first step be addressed?

If you are someone associated with OSMand app who has some insight into the governance / decision making around which problems gets fixed, please put your hand up. 

This is a major barrier to usability for people looking for an OSM based app. Decent search functionality is a pretty basic and reasonable expectation of a digital mapping app.

**Reply by kannes** [reply]
That sounds a bit aggressive and entitled. Do you have monetary funding for developer time on this or can you contribute yourself?

**Reply by sarahsquirrel** [reply]
Yes, very willing to pitch in to make it happen.
What's the best way in to discuss that?

**Reply by Sonwon1** [reply]
> That sounds a bit aggressive and entitled. Do you have monetary funding for developer time on this or can you contribute yourself?

I agree that it does sound aggressive, not with entitled. I think it is just frustration since this request was opened in 2023. And the search engine has been problematic before even 2023.

**Reply by sarahsquirrel** [reply]
I'd welcome your suggestions for other approaches, to get traction on the issue

**Reply by vshcherb** [reply]
In 5.2 and 5.3 major issues were fixed: 5.4 release is focused to reduce search results to make it more efficients. 2 issues are planned for next steps: streets with space  & combined search (<Poi type | poi by name> <City | Town | Admin Boundary>)

Please make sure that your specific non working query is mentionned in one of the issues! Otherwise this is not a query rather emotional comment which will not be taken into account.
